### PR TITLE
Antiforgery goes at the end of filters

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes aspnet/Security#1009. Since Antiforgery should be at the end of filters by default set its order to be a large number by default.

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
:up::date:

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/AutoValidateAntiforgeryTokenAttribute.cs [hunk 1]
```diff
@@ -17,12 +17,14 @@ namespace Microsoft.AspNetCore.Mvc
     /// validation of antiforgery tokens by default for an application. Use
     /// <see cref="IgnoreAntiforgeryTokenAttribute"/> to suppress validation of the antiforgery token for
     /// a controller or action.
+    /// The default Order for this attribute is 1000 because it must run after any filter which does authentication
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/AutoValidateAntiforgeryTokenAttribute.cs]: I would suggest putting this document on the order property. Also when you're working on doc comments, newlines in the source code are not significant. Use the `<para></para>` tag if you want to make paragraphs

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/AntiforgeryAuthTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/AntiforgeryAuthTests.cs]: Header

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/project.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "warningsAsErrors": true
   },
   "dependencies": {
+    "AjaxAntiForgeryValidation": "1.0.0",
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/project.json]: Suggest calling this SecurityWebSite

File: test/WebSites/AjaxAntiForgeryValidation/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Add framework services.
+            services.AddMvc();
+            services.AddAntiforgery(options =>
+            {
+                Console.WriteLine();
+            });
+            services.AddIdentity<ApplicationUser, ApplicationRole>(config =>
```
Comment (rynowak) [inline test/WebSites/AjaxAntiForgeryValidation/Startup.cs]: Do we need Identity here? This is pretty heavyweight. Identity is primarily concerned which information about the user, not just whether or not the user is authentic.

Suggest talking to @Tratcher about how we we could just get away with cookie middleware (which Identity uses).

File: test/WebSites/AjaxAntiForgeryValidation/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        public IConfigurationRoot Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Add framework services.
+            services.AddMvc();
+            services.AddAntiforgery(options =>
+            {
+                Console.WriteLine();
```
Comment (rynowak) [inline test/WebSites/AjaxAntiForgeryValidation/Startup.cs]: derp

File: test/WebSites/AjaxAntiForgeryValidation/Startup.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+            Configuration = builder.Build();
+        }
+
+        public IConfigurationRoot Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
```
Comment (rynowak) [inline test/WebSites/AjaxAntiForgeryValidation/Startup.cs]: Don't need all this silliness

File: test/WebSites/AjaxAntiForgeryValidation/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+            app.Use(next => context =>
+            {
+                var antiforgery = context.RequestServices.GetService<IAntiforgery>();
+
+                if (context.Request.Path.Value.ToLower().Equals("/") ||
+                    context.Request.Path.Value.ToLower().StartsWith("/home"))
+                {
+                    var tokens = antiforgery.GetAndStoreTokens(context);
+                    context.Response.Cookies.Append("XSRF-TOKEN", tokens.RequestToken, new CookieOptions() { HttpOnly = false });
```
Comment (Tratcher) [inline test/WebSites/AjaxAntiForgeryValidation/Startup.cs]: Why is this here?

**Comment by ryanbrandenburg** [comment]
:up::date:

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/AutoValidateAntiforgeryTokenAttribute.cs [hunk 8]
```diff
@@ -22,7 +22,11 @@ namespace Microsoft.AspNetCore.Mvc
     public class AutoValidateAntiforgeryTokenAttribute : Attribute, IFilterFactory, IOrderedFilter
     {
         /// <inheritdoc />
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/AutoValidateAntiforgeryTokenAttribute.cs]: Inheritdoc doesn't really work with these kinds of thing. you'll want to also provide a summary here

**Comment by ryanbrandenburg** [comment]
:up::date:
